Assets/Scripts/GameManagement/DataPersistence/FileDataHandler.cs

```diff
@@ -0,0 +1,114 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System;
+using System.IO;
+using UnityEngine.Playables;
+using UnityEngine.Profiling;
+
+public class FileDataHandler
+{
+    private string dataDirPath = "";
+    private string dataFileName = "";
+    private string fullPath = "";
+
+    private FileStream fileStream;
+    private StreamWriter fileWriter;
+    private StreamReader fileReader;
+
+    public FileDataHandler(string dataDirPath, string dataFileName)
+    {
+        this.dataDirPath = dataDirPath;
+        this.dataFileName = dataFileName;
+    }
+
+    //Open file methods
+    public void OpenFileSave()
+    {
+        fullPath = Path.Combine(dataDirPath, dataFileName);
+
+        try
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(fullPath));
+            fileStream = new FileStream(fullPath, FileMode.Create);
+            fileWriter = new StreamWriter(fileStream, System.Text.Encoding.UTF8, 64, true);
+        }
+        catch (Exception e)
+        {
+            Debug.LogError("Error occured when trying to open file: " + fullPath + "\n" + e);
+        }
+    }
+
+    public void OpenFileLoad()
+    {
+        fullPath = Path.Combine(dataDirPath, dataFileName);
+
+        try
+        {
+            Directory.CreateDirectory(Path.GetDirectoryName(fullPath));
```

Comment:
Does the path need to be created if you are trying to read from it?

Reply:
no, good catch